File: manifest_updater.py
import os
import json
from pathlib import Path

def generate_manifest(script_dir="scripts", output_file="manifest.json"):
    """
    scripts 디렉토리 내의 .js 파일들을 찾아 manifest.json을 생성합니다.
    """
    p = Path(script_dir)
    if not p.exists():
        print(f"Error: Directory '{script_dir}' not found.")
        return

    scripts = [f.name for f in p.glob("*.js")]
    manifest = {"files": scripts}

    with open(output_file, "w", encoding="utf-8") as f:
        json.dump(manifest, f, indent=4)
    
    print(f"✅ Generated {output_file} with {len(scripts)} scripts.")

if __name__ == "__main__":
    # 기본적으로 Bitburner 루트에서 실행된다고 가정
    script_path = Path(__file__).parent / "scripts"
    if not script_path.exists():
        script_path.mkdir(exist_ok=True)
        # 기본 템플릿 파일 생성
        (script_path / "template.js").write_text("/** @param {NS} ns */\nexport async function main(ns) {\n    ns.tprint('Hello from Bitburner Assistant!');\n}")
        print("📁 Created scripts directory and template.js")

    generate_manifest(script_path, Path(__file__).parent / "manifest.json")

File: manifest_updater.js
const fs = require('fs');
const path = require('path');

/**
 * scripts 디렉토리 내의 .js 파일들을 찾아 manifest.json을 생성합니다.
 */
function generateManifest() {
    const scriptDir = path.join(__dirname, 'scripts');
    const outputFile = path.join(__dirname, 'manifest.json');

const manifestPath = path.join(__dirname, 'manifest.json');

// euoom 폴더 내부의 모든 .js 파일을 재귀적으로 찾음
function scanDirectory(dir, fileList = []) {
    const files = fs.readdirSync(dir);
    files.forEach(file => {
        const filePath = path.join(dir, file);
        const stat = fs.statSync(filePath);
        if (stat.isDirectory()) {
            scanDirectory(filePath, fileList);
        } else if (file.endsWith('.js')) {
            // 프로젝트 루트 기준의 경로로 변환 (예: euoom/analyze/scan.js)
            const relativePath = path.relative(__dirname, filePath).replace(/\\/g, '/');
            fileList.push(relativePath);
        }
    });
    return fileList;
}

try {
    const scriptsDir = path.join(__dirname, 'euoom');
    
    // euoom 디렉토리가 없으면 생성
    if (!fs.existsSync(scriptsDir)) {
        console.log(`📁 Creating euoom directory...`);
        fs.mkdirSync(scriptsDir, { recursive: true });
        // 기본 템플릿 파일 생성
        const templatePath = path.join(scriptsDir, 'template.js');
        if (!fs.existsSync(templatePath)) {
            fs.writeFileSync(templatePath, "/** @param {NS} ns */\nexport async function main(ns) {\n    ns.tprint('Hello from Bitburner Assistant!');\n}");
        }
    }

    const scripts = scanDirectory(scriptsDir);
        
        console.log(`✅ Generated manifest.json with ${scripts.length} scripts.`);
    } catch (error) {
        console.error(`❌ Error scanning scripts: ${error.message}`);
    }
}

generateManifest();
